<script>
    Promise.all = (promises) =>{
        let responses = [];
        let errorResp = [];
      return new Promise ((resolve,reject)=>{
         promises.forEach(async(singlePromise,i)=>{
            try{
             let res = await singlePromise;
             responses.push(res);
             if(i == promises.length-1){
                 if(errorResp.length>0){
                    reject(errorResp);
                 }
                 else{
                    resolve("custom promise::" + responses);
                 }
             }
            }catch(err){
                      errorResp.push(err);
                      reject(err);
            }
         })
      })
    }

    let p1 = Promise.resolve("Promise1 resolved");

    let p2 = new Promise((resolve,reject)=>{
        setTimeout(()=>{
            resolve("promise2 is in resolved");
        },1000)
    });

    Promise.all([p1,p2]).then((res)=>{
        console.log(res);
    },(err)=>{
        console.log("error",err);
    })
</script>